lib/list_versions.go

```diff
@@ -102,7 +102,7 @@ func GetTFReleases(mirrorURL string, preRelease bool) ([]*Release, error) {
 	limit := 20
 	u, err := url.Parse(mirrorURL)
 	if err != nil {
-		return nil, fmt.Errorf("Error parsing url: %s", err)
+		return nil, fmt.Errorf("[Error] : parsing url: %s", err)
```

Comment:
This is not essential, though, if I got things right, when using `%q` (instead of `%s`) we can get properly quoted strings, which may help to identify start and end of the original error message per se to visually abstract it from fancy message we put right before it. Leaving this for your consideration since this doesn't bring much of benefit, but rather somewhat more of a readability for end users.

Reply:
I'm away for the weekend and didn't want to be on my computer (on mobile now, even pushed the last commit from my phone haha) too much so I'd rather address some of the cleanup and improvements in a follow up PR if you don't see it as a blocker.
That or if it's not reviewed by Tuesday I'll add these in.

Reply:
Update with https://github.com/warrensbox/terraform-switcher/commit/7346f8d07bbf111c65171769e9eec3b13839b8cb

Reply:
Go ahead. I will review this over the weekend (May 21 and 22)

Reply:
Updated with some additional error formatting suggestions. Please take another look @yermulnik if this is sufficient.

Reply:
@Jukie This is awesome huge piece of work! Thank you.